src/main/resources/static/js/MainWindow.js

```diff
@@ -1,670 +1,769 @@
-    showstep1();
+	document.getElementById("delimiter").value = ";";
+	document.getElementById("headerlines").value = "0";
+	$("#things").val(null).trigger("change");
+	$("#selecttime").val(null).trigger("change");
+	$("#timeTable").find("tbody").find("tr").each(function() {
+		this.remove();
+	});
+	addRow($("#timeTable"), "2", "20");
+	document.getElementById("datastreams").innerHTML = "";
+	$("#configs").val(null).trigger("change");
+	resetMapping();
+
+	optimizeforsource(); // disables the delimiter and time format if the source
+	// file is an excel file
+	showstep1();
 
 }
 
 /**
  * gets all things from the frost-server
  */
 function getThings(fnSuccess) {
-	
-    var listTh = $("#things");
-    var list2 = $("#datastreams");
-    
-    listTh.empty().append(new Option("", "", null, null));
-    list2.empty().append(new Option("", "", null, null));
-    
-    listTh.trigger("change");
-    list2.trigger("change");
-    
-	
+
+	var listTh = $("#things");
+	var list2 = $("#datastreams");
+
+	listTh.empty().append(new Option("", "", null, null));
+	list2.empty().append(new Option("", "", null, null));
+
+	listTh.trigger("change");
+	list2.trigger("change");
+
 	var url = document.getElementById("serverurlbox").innerText;
 	var mydata = {
-		frostUrlString: url	
+		frostUrlString : url
 	};
-    $.ajax({
-        type: "GET",
-        url: "thing/all",
-        data: mydata,
-        success: function (response) {
-            var json = JSON.stringify(response, null, 4);
-            var jsonparsed = JSON.parse(json);
-
-            var list = $("#things");
-            list.empty().append(new Option("", "", null, null));
-            for (var i = 0; i < jsonparsed.length; i++) {
-                var option = new Option(jsonparsed[i].name + " (" + jsonparsed[i].frostId + ")", jsonparsed[i].name + " (" + jsonparsed[i].frostId + ")", null, null);
-                option.setAttribute("data-value", JSON.stringify(jsonparsed[i], null, 4));
-                list.append(option);
-            }
-            list.trigger("change");
-            addToLog("Things loaded.");
-            
-            if (!(fnSuccess == null)) {
-        		fnSuccess(); 
-        	}
-        },
-        error: function (e) {
-            addToLog(e.responseText);
-            
-        }
-    });
-    
-}
+	$.ajax({
+		type : "GET",
+		url : "thing/all",
+		data : mydata,
+		success : function(response) {
+			var json = JSON.stringify(response, null, 4);
+			var jsonparsed = JSON.parse(json);
+
+			var list = $("#things");
+			list.empty().append(new Option("", "", null, null));
+			jsonparsed.forEach(function(val) {
+				var option = new Option(val.name + " ("
+						+ val.frostId + ")", val.name
+						+ " (" + val.frostId + ")", null, null);
+				option.setAttribute("data-value", JSON.stringify(val,
+						null, 4));
+				list.append(option);
+			});
+			
+			list.trigger("change");
+			addToLog("Things loaded.");
+
+			if (!(fnSuccess == null)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FraunhoferIOSB/ChillImport/pullRequest?prid=4781409)